packages/local_auth/local_auth_ios/ios/Classes/FLTLocalAuthPlugin.m

```diff
@@ -49,161 +75,128 @@ - (instancetype)initWithContextFactory:(NSObject<FLAAuthContextFactory> *)factor
-                 withFlutterResult:(FlutterResult)result {
-  LAContext *context = [self.authContextFactory createAuthContext];
-  NSError *authError = nil;
-  self.lastCallArgs = nil;
-  self.lastResult = nil;
-  context.localizedFallbackTitle = arguments[@"localizedFallbackTitle"] == [NSNull null]
-                                       ? nil
-                                       : arguments[@"localizedFallbackTitle"];
-
-  if ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics
-                           error:&authError]) {
-    [context evaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics
-            localizedReason:arguments[@"localizedReason"]
-                      reply:^(BOOL success, NSError *error) {
-                        dispatch_async(dispatch_get_main_queue(), ^{
-                          [self handleAuthReplyWithSuccess:success
-                                                     error:error
-                                          flutterArguments:arguments
-                                             flutterResult:result];
-                        });
-                      }];
-  } else {
-    [self handleErrors:authError flutterArguments:arguments withFlutterResult:result];
-  }
+- (nullable NSNumber *)isDeviceSupportedWithError:
+    (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  // TODO(stuartmorgan): Fix this to check for biometrics or passcode; see
+  // https://github.com/flutter/flutter/issues/116179
+  return @YES;
 }
 
-- (void)authenticate:(NSDictionary *)arguments withFlutterResult:(FlutterResult)result {
-  LAContext *context = [self.authContextFactory createAuthContext];
-  NSError *authError = nil;
-  _lastCallArgs = nil;
-  _lastResult = nil;
-  context.localizedFallbackTitle = arguments[@"localizedFallbackTitle"] == [NSNull null]
-                                       ? nil
-                                       : arguments[@"localizedFallbackTitle"];
+#pragma mark Private Methods
 
-  if ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthentication error:&authError]) {
-    [context evaluatePolicy:kLAPolicyDeviceOwnerAuthentication
-            localizedReason:arguments[@"localizedReason"]
-                      reply:^(BOOL success, NSError *error) {
-                        dispatch_async(dispatch_get_main_queue(), ^{
-                          [self handleAuthReplyWithSuccess:success
-                                                     error:error
-                                          flutterArguments:arguments
-                                             flutterResult:result];
-                        });
-                      }];
-  } else {
-    [self handleErrors:authError flutterArguments:arguments withFlutterResult:result];
+- (void)showAlertWithMessage:(NSString *)message
+                 firstButton:(NSString *)firstButton
+            additionalButton:(NSString *)secondButton
```

Comment:
nit: `dismissButtonTitle` and `openSettingsButtonTitle`

Reply:
Done; I hadn't really processed the rest of the signature since I was focused on fixing the really terrible first part :)